bedwars-plugin/src/main/java/com/tomkeuper/bedwars/addon/AddonManager.java

```diff
@@ -2,44 +2,85 @@
+import com.tomkeuper.bedwars.api.addon.AddonStorer;
 import com.tomkeuper.bedwars.api.addon.IAddonManager;
-import java.util.ArrayList;
+import org.bukkit.Bukkit;
+
 import java.util.List;
 import java.util.stream.Collectors;
 
-public class AddonManager implements IAddonManager {
-    private static List<Addon> addons;
+public class AddonManager extends AddonStorer implements IAddonManager {
+    private static List<Addon> registeredAddons;
+    private static List<Addon> unloadedAddons;
+    private static List<Addon> loadedAddons;
 
     public AddonManager() {
-        if (addons == null) addons = new ArrayList<>();
+        registeredAddons = registeredAddons();
+        unloadedAddons = unloadedAddons();
+        loadedAddons = loadedAddons();
     }
+
+
     @Override
     public List<Addon> getAddons() {
-        return addons;
+        return registeredAddons;
+    }
+
+    @Override
+    public List<Addon> getLoadedAddons() {
+        return loadedAddons;
+    }
+
+    @Override
+    public List<Addon> getUnloadedAddons() {
+        return unloadedAddons;
     }
 
     @Override
     public List<Addon> getAddonsByAuthor(String author) {
-        return addons.stream().filter(a -> a.getAuthor().equals(author)).collect(Collectors.toList());
+        return registeredAddons.stream().filter(a -> a.getAuthor().equals(author)).collect(Collectors.toList());
     }
 
     @Override
     public void loadAddon(Addon addon) {
-        addons.add(addon);
+        loadedAddons.add(addon);
+        if (!Bukkit.getPluginManager().isPluginEnabled(addon.getPlugin()))
+            Bukkit.getPluginManager().enablePlugin(addon.getPlugin());
     }
 
-    public void registerAddons() {
+    @Override
+    public void unloadAddon(Addon addon) {
+        unloadedAddons.add(addon);
+        addon.unload();
+        Bukkit.getPluginManager().disablePlugin(addon.getPlugin());
+    }
+
+    @Override
+    public void unloadAddons() {
+        if (loadedAddons.isEmpty()) return;
+        log("Unloading addons...");
+        for (Addon addon : loadedAddons) {
+            log("Unloading " + addon.getIdentifier() + " by " + addon.getAuthor());
+            addon.unload();
+            Bukkit.getPluginManager().disablePlugin(addon.getPlugin());
+            log("Addon unloaded successfully!");
+        }
+    }
+
+    @Override
+    public void loadAddons() {
         String count = "";
-        if (addons.size() < 1) {
+        if (registeredAddons.size() < 1) {
             log("No addons were found!");
             return;
         }
-        else if (addons.size() == 1) count = "addon";
-        else if (addons.size() > 1) count = "addons";
-        log(addons.size() + " " + count + " has been found!");
-        log("Loading " + addons.size() + " " + count);
-        for (Addon addon : addons) {
+        else if (registeredAddons.size() == 1) count = "addon";
+        else if (registeredAddons.size() > 1) count = "addons";
```

Comment:
quite a vouch for this haha, most people will just add "addon(s)"

Reply:
haha thats true, but i preffer it to look like that, not sure what other people will think